hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -638,6 +666,11 @@ def _networks(bits):
     type(Ellipsis): st.just(Ellipsis),
     type(NotImplemented): st.just(NotImplemented),
     bytearray: st.binary().map(bytearray),
+    # this is a pragmatic decision and not strictly correct. memoryview is
+    # generic in its element type, which is not necessarily bytes. However,
+    # most usages are memoryview[bytes].
+    #
+    # See https://docs.python.org/3/library/stdtypes.html#memoryview.
     memoryview: st.binary().map(memoryview),
```

Comment:
It looks like `memoryview` is _conceptually_ generic, but doesn't actually implement any of the machinery we'd need for that to work at runtime - you can't even execute `memoryview[bytes]`, for example!  If that was fixed, we could handle `from_type(memoryview[T])` as `from_type(list[T]).map(buffer_proto_impl).map(memoryview)`.

This feels like it might be worth an upstream issue, but it's pretty low-priority for me.

Reply:
Works for me!

```python
 ~/Desktop λ python3.14
Python 3.14.0rc1 (main, Jul 23 2025, 00:27:39) [Clang 20.1.4 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> memoryview[bytes]
memoryview[bytes]
>>> class A:
...     pass
...     
>>> memoryview[A]
memoryview[__main__.A]
>>> 
```

Reply:
🤦‍♂️ I was on Python 3.12, oops.

Want to add the registration machinery then, as we do for other generics?

Reply:
I looked into it and I'd like to treat buffers more comprehensively by also implementing `collections.abc.Buffer` from https://peps.python.org/pep-0688/ (3.12+) - let's defer to a future issue + pr?